pkg/config/config.go

```diff
@@ -4,17 +4,36 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"reflect"
 	"strings"
 
-	"github.com/bacalhau-project/bacalhau/pkg/config/types"
 	"github.com/mitchellh/mapstructure"
 	"github.com/spf13/viper"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/configenv"
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
 )
 
+var _ Context = (*config)(nil)
+
+type Context interface {
```

Comment:
the name `Context` is confusing to me. I keep relating it to the context we pass around methods, and I thought this interface extends `context.Context`

Reply:
I can rename this type to something different, suggestions welcome.
I picked this name because `config.Context` felt descriptive (it's the context of the current configuration) and didn't stutter like `config.Config`